lib: tenstorrent: bh_arc: add error logging to spi_eeprom

danieldegrasse opened the pull request:
Add error logging to SPI eeprom for instances where flash read or write fails.

cfriedt commented on lib/tenstorrent/bh_arc/spi_eeprom.c:
```diff
@@ -71,17 +81,20 @@ static int SpiSmartWrite(uint32_t address, const uint8_t *data, uint32_t num_byt
 		}
 		rc = flash_write(flash, addr, spi_page_buf, sector_size);
 		if (rc < 0) {
+			LOG_ERR("Flash write failed [phase 1] at 0x%08x: %d", addr, rc);
```
```suggestion
		LOG_ERR("Flash %s failed %sat 0x%08x: %d", "write", "[phase 1] ", addr, rc);
```

cfriedt commented on lib/tenstorrent/bh_arc/spi_eeprom.c:
```diff
@@ -94,16 +107,19 @@ static int SpiSmartWrite(uint32_t address, const uint8_t *data, uint32_t num_byt
 
 		rc = flash_read(flash, addr, spi_page_buf, sector_size);
 		if (rc < 0) {
+			LOG_ERR("Flash read failed [phase 2] at 0x%08x: %d", addr, rc);
```
```suggestion
		LOG_ERR("Flash %s failed %sat 0x%08x: %d", "read", "[phase 2] ", addr, rc);
```

cfriedt commented on lib/tenstorrent/bh_arc/spi_eeprom.c:
```diff
@@ -94,16 +107,19 @@ static int SpiSmartWrite(uint32_t address, const uint8_t *data, uint32_t num_byt
 			}
 			rc = flash_write(flash, addr, data, sector_size);
 			if (rc < 0) {
+				LOG_ERR("Flash write failed [phase 2] at 0x%08x: %d", addr, rc);
```
```suggestion
		LOG_ERR("Flash %s failed %sat 0x%08x: %d", "write", "[phase 2] ", addr, rc);
```

cfriedt commented on lib/tenstorrent/bh_arc/spi_eeprom.c:
```diff
@@ -121,17 +137,20 @@ static int SpiSmartWrite(uint32_t address, const uint8_t *data, uint32_t num_byt
 		memcpy(spi_page_buf, data, chunk_size);
 		rc = flash_erase(flash, addr, sector_size);
 		if (rc < 0) {
+			LOG_ERR("Flash erase failed [phase 3] at 0x%08x: %d", addr, rc);
```
```suggestion
		LOG_ERR("Flash %s failed %sat 0x%08x: %d", "erase", "[phase 3] ", addr, rc);
```

danieldegrasse commented:
> Still quite a few strings that can be deduplicated.

Where? I believe I addressed your last round of comments if you are referring to those

danieldegrasse commented:
> > > Still quite a few strings that can be deduplicated.
> > 
> > 
> > Where? I believe I addressed your last round of comments if you are referring to those
> 
> They're all of the comments that are still unresolved.
> 
> [#586 (comment)](https://github.com/tenstorrent/tt-zephyr-platforms/pull/586#discussion_r2346751807) [#586 (comment)](https://github.com/tenstorrent/tt-zephyr-platforms/pull/586#discussion_r2375565070) [#586 (comment)](https://github.com/tenstorrent/tt-zephyr-platforms/pull/586#discussion_r2346754000) [#586 (comment)](https://github.com/tenstorrent/tt-zephyr-platforms/pull/586#discussion_r2346754334) [#586 (comment)](https://github.com/tenstorrent/tt-zephyr-platforms/pull/586#discussion_r2346754553) [#586 (comment)](https://github.com/tenstorrent/tt-zephyr-platforms/pull/586#discussion_r2346754935) [#586 (comment)](https://github.com/tenstorrent/tt-zephyr-platforms/pull/586#discussion_r2346755242) [#586 (comment)](https://github.com/tenstorrent/tt-zephyr-platforms/pull/586#discussion_r2346755456) [#586 (comment)](https://github.com/tenstorrent/tt-zephyr-platforms/pull/586#discussion_r2346755941) [#586 (comment)](https://github.com/tenstorrent/tt-zephyr-platforms/pull/586#discussion_r2346756282)

Fixed now
